alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -138,51 +141,34 @@ transAddr :: Addr crypto -> Maybe P.Address
 transAddr (Addr _net object stake) = Just (P.Address (transCred object) (transStakeReference stake))
 transAddr (AddrBootstrap _bootaddr) = Nothing
 
--- ===============================
--- Translate ValidityIntervals
--- TODO remove this function when we are using transVITime instead
-transVI ::
+slotToPOSIXTime ::
   EpochInfo Identity ->
   SystemStart ->
-  ValidityInterval ->
-  P.SlotRange
-transVI _ _ (ValidityInterval SNothing SNothing) =
-  P.Interval
-    (P.LowerBound P.NegInf True)
-    (P.UpperBound P.PosInf False)
-transVI _ _ (ValidityInterval (SJust (SlotNo i)) SNothing) =
-  P.Interval
-    (P.LowerBound (P.Finite (fromIntegral i)) True)
-    (P.UpperBound P.PosInf False)
-transVI _ _ (ValidityInterval SNothing (SJust (SlotNo i))) =
-  P.Interval
-    (P.LowerBound P.NegInf True)
-    (P.UpperBound (P.Finite (fromIntegral i)) False)
-transVI _ _ (ValidityInterval (SJust (SlotNo i)) (SJust (SlotNo j))) =
-  P.Interval
-    (P.LowerBound (P.Finite (fromIntegral i)) True)
-    (P.UpperBound (P.Finite (fromIntegral j)) False)
+  SlotNo ->
+  P.Extended P.POSIXTime
+slotToPOSIXTime ei sysS s =
+  P.Finite . P.POSIXTime . resolution . nominalDiffTimeToSeconds . utcTimeToPOSIXSeconds . runIdentity $
```

Comment:
Is `resolution . nominalDiffTimeToSeconds` the best way to convert ` NominalDiffTime` to an integer?

Reply:
Looks reasonable to me... I guess Plutus doesn't want to use the actual `Pico` type?

Reply:
@michaelpj we could give y'all [Pico](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Fixed.html#t:Pico) instead of Integer if you'd like.

Reply:
Don't want it. We decided to keep things dumb. (cc @j-mueller)

Reply:
Sounds reasonable to me. Kronecker wasn't all wrong!